platform/util/uuidgenerator/platform-util-uuidgenerator-api/src/main/java/org/codice/ddf/platform/util/uuidgenerator/MetacardUuidGenerator.java

```diff
@@ -0,0 +1,40 @@
+/**
+ * Copyright (c) Codice Foundation
+ * <p>
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser
+ * General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or any later version.
+ * <p>
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.platform.util.uuidgenerator;
+
+/**
+ * <p>
+ * <b> This code is experimental. While this interface is functional and tested, it may change or be
+ * removed in a future version of the library. </b>
+ * </p>
+ *
+ * The {@code MetacardUuidGenerator} generates {@link java.util.UUID}s for Metacards
+ */
+public interface MetacardUuidGenerator {
```

Comment:
Could we take advantage of some of our built in java interfaces and make this interface `extend Supplier<String>` that way we can use it as a drop in supplier? (And then the overriden `get` method could just call `getMetacardUuid`)